### DISCUSSION TITLE
File upload with additional form-data results in UnprocessableEntity[422]

### DISCUSSION THREAD

**Discussion by raghav-knowbe4** [fastapi/fastapi#10966] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, File, Form, UploadFile

app = FastAPI()

@app.post("/upload")
async def upload_files(file: UploadFile = File(...), context: str = Form(...)):
    return {"hello": "world"}
```

### Description

I am able to access the file with the following implementation:
```
@app.post("/upload")
async def upload_files(file: UploadFile = File(...)):
    print(file.filename) # this works
    return {"hello": "world"}
```

However, when I send additional form-data along with the file, there are two cases:
```
@app.post("/upload")
async def upload_files(file: UploadFile = File(...), context: str = Form(...)):
    return {"hello": "world"}
```

With "Content-Length" in headers, I get following error:
```
Did not find boundary character 99 at index 2
```

Without "Content-Length" in headers, I get UnprocessableEntity[422] with following response:
```
{
    "detail": [
        {
            "type": "missing",
            "loc": [
                "body",
                "context"
            ],
            "msg": "Field required",
            "input": null,
            "url": "https://errors.pydantic.dev/2.5/v/missing"
        },
        {
            "type": "missing",
            "loc": [
                "body",
                "file"
            ],
            "msg": "Field required",
            "input": null,
            "url": "https://errors.pydantic.dev/2.5/v/missing"
        }
    ]
}
```

### Operating System

macOS

### FastAPI Version

0.109.0

### Pydantic Version

2.5.3

### Python Version

3.11

**Comment by Kludex** [comment]
Can you share the client call you are using, and which client, please?

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
